services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceParentHelixAdmin.java

```diff
@@ -1460,7 +1460,8 @@ public Version incrementVersionIdempotent(
       boolean isExistingPushJobARepush = Version.isPushIdRePush(existingPushJobId);
       boolean isIncomingPushJobARepush = Version.isPushIdRePush(pushJobId);
 
-      if (getLingeringStoreVersionChecker()
+      // If version swap is enabled, do not check for lingering push as user may swap at much later time.
+      if (!version.isVersionSwapDeferred() && getLingeringStoreVersionChecker()
```

Comment:
IIUC the expected behavior after the change is that:
1. Old push will be prevented from getting killed automatically since it will not be picked up by the lingering store version checker.
2. User will get the concurrent push error message and needs to either complete the swap of previous version or kill the previous push to proceed.

Questions:
After the change it still throws a generic concurrent push error message. Do users have enough information to take the right action? Should we add a bit more code to differentiate scenarios for our users? The parent controller should have enough information to determine if the previous push is waiting for version swap or not, right?

Reply:
Right, your understanding is correct. I could update the message in concurrent exception to indicate what exactly is going on. I have added the following log in parent controller, but that was not useful as vpj wont be showing it. 
`"There is already future version {} exists for store {}, please wait till the future version is made current."`

Reply:
I see, at least it will leave some trace in logs and help the operator figure out what happened.